SwissKnife/gcloud/GCloudStorage.py

```diff
@@ -0,0 +1,159 @@
+        """Uploads data, which may come in several formats, to
+        Storage. Accepted data types are 'file' and 'str', matching
+        methods save_file and save_string of this same class.
+
+        :param data: Data file or string that will be uploaded
+        :type data: [type]
+        :param data_encoding: Specific encoding of the file
+        :type data_encoding: str
+        :param data_type: Type of data to upload: file or string
+        :type data_type: str
+        :param destionation_storage_path: Storage path were it'll be stored
+        :type destionation_storage_path: str
+        :param destionation_file_name: Name that it will have in Storage
+        :type destionation_file_name: str
+        :raises NotImplementedError: When the data type provided is not implemented
+        :return: gcloud storage of the uploaded file
+        :rtype: str
+        """
+        blob = self.__generate_blob(destionation_storage_path, destionation_file_name, data_encoding)
+        
+        if type == 'str':
+            blob.upload_from_string(data)
+        elif type == 'file':
+            blob.upload_from_filename(data)
+        else:
+            raise NotImplementedError
+        
+        file_path = self._get_storage_file_path(destionation_storage_path, destionation_file_name, with_bucket=True)
+        
+        self.logger.info(f"Uploaded file to gcloud path {file_path}")
+        return file_path
+        
+    def __generate_blob(self, 
+                        file_path: str,
+                        file_name: str,
+                        encoding: str):
+        blob_name = self.get_storage_complete_file_path(file_path, file_name)
+        blob = self.bucket.blob(blob_name)
+        
+        if encoding:
+            blob.content_encoding = 'gzip'
+            
+        return blob
+    
+    @staticmethod
+    def get_storage_complete_file_path(file_path: str,
+                                       file_name: str,
+                                       with_bucket: bool=False,
+                                       with_prefix: bool=True) -> str:
+        """Returns the complete path of a file stored in gcloud.
+
+        :param file_path: Path of the file, without bucket
+        :type file_path: str
+        :param file_name: Name of the file
+        :type file_name: str
+        :param with_bucket: If the bucket will be returned, defaults to False
+        :type with_bucket: bool, optional
+        :param with_prefix: If prefix set by BUCKET_PATH_PREFIX is used, defaults to True
+        :type with_prefix: bool, optional
+        :return: Complete path of a file stored in gcloud
+        :rtype: str
+        """
+        bucket_name = f'{BUCKET_NAME}/' if with_bucket else ''
+        file_prefix = f'{BUCKET_PATH_PREFIX}/' if with_prefix and BUCKET_PATH_PREFIX else ''
+        
+        return f'{bucket_name}{file_prefix}{file_path}/{file_name}'
+
+    def list_blobs(self, storage_path: str):
```

Comment:
The return type is missing.

Reply:
Yeah, it returns an iterator and I was wondering how to annotate it, I'll use the trick mentioned below.